### ISSUE TITLE
🐛 [firebase_messaging] localized notifications crash android app

### ISSUE THREAD

**Issue by pawsong** [FirebaseExtended/flutterfire#4072]
## Bug report

**Describe the bug**
Localized notifications crash android app.

When `firebase_messaging` passes `RemoteMessage` object to Dart land, the object passed to Dart land contains string array values for `titleLocArgs` and `bodyLocArgs` keys. ([link](https://github.com/FirebaseExtended/flutterfire/blob/7d91938fb9a78078f658ec099bc72be6ecd08773/packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingUtils.java#L92)).

The flutter engine doesn't seem to support string array ([link](https://github.com/flutter/engine/blob/48687ca3ca6e4f24281ea998789cc21ad840accb/shell/platform/android/io/flutter/plugin/common/StandardMessageCodec.java#L276)), so crash happens.

<img width="927" alt="스크린샷 2020-11-09 오후 11 30 31" src="https://user-images.githubusercontent.com/2945463/98553993-9228cc80-22e3-11eb-9503-7ce2edda1d51.png">

Not sure if ios has similar issues or not.

### Steps to reproduce

1. Send localized notification from server (with `titleLocArgs` or `bodyLocArgs` fields).
2. See crash

### Expected behavior

Localized notifications does not crash android app.

---

## Additional context

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel stable, 1.22.3, on Mac OS X 10.15.5 19F101, locale ko-KR)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
[✓] Android Studio (version 4.0)
[✓] VS Code (version 1.50.1)
[✓] Connected device (1 available)
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.10.3
Flutter SDK 1.22.3

dependencies:
- cached_network_image 2.3.2+1 [flutter flutter_cache_manager octo_image]
- carousel_slider 2.2.1 [flutter]
- confetti 0.5.4+1 [flutter random_color vector_math]
- cupertino_icons 1.0.0
- equatable 1.2.5 [collection meta]
- firebase_analytics 6.0.0 [meta flutter firebase_core firebase_analytics_web firebase_analytics_platform_interface]
- firebase_core 0.5.2 [firebase_core_platform_interface flutter quiver meta firebase_core_web]
- firebase_crashlytics 0.2.3 [flutter stack_trace firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface]
- firebase_messaging 8.0.0-dev.7 [meta flutter firebase_core firebase_core_platform_interface firebase_messaging_platform_interface]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_app_badger 1.1.2 [flutter]
- flutter_icons 1.1.0 [flutter]
- flutter_local_notifications 3.0.1+2 [flutter platform flutter_local_notifications_platform_interface timezone]
- flutter_localizations 0.0.0 [flutter intl characters collection meta path typed_data vector_math]
- flutter_login_facebook 0.4.0+1 [flutter list_ext]
- flutter_slidable 0.5.7 [flutter]
- flutter_svg 0.19.0 [path_drawing xml vector_math meta flutter]
- fluttertoast 7.1.0 [flutter flutter_web_plugins]
- google_sign_in 4.5.3 [google_sign_in_platform_interface flutter meta google_sign_in_web]
- graphql_flutter 3.1.0 [graphql flutter meta path path_provider rxdart connectivity gql]
- http 0.12.2 [http_parser path pedantic]
- image_cropper 1.3.1 [flutter]
- image_picker 0.6.7+7 [flutter flutter_plugin_android_lifecycle image_picker_platform_interface]
- in_app_review 1.0.2 [flutter in_app_review_platform_interface]
- intl 0.16.1 [path]
- intl_translation 0.17.10+1 [analyzer args dart_style intl path petitparser]
- mime_type 0.3.2
- modal_bottom_sheet 0.2.0+1 [flutter]
- package_info 0.4.3 [flutter]
- path_provider 1.6.14 [flutter path_provider_platform_interface path_provider_macos path_provider_linux]
- provider 4.3.2+2 [flutter nested collection]
- pull_to_refresh 1.6.1 [flutter]
- share 0.6.5 [meta mime flutter]
- shared_preferences 0.5.10 [meta flutter shared_preferences_platform_interface shared_preferences_linux shared_preferences_macos shared_preferences_web]
- sign_in_with_apple 2.5.2 [meta flutter]
- super_tooltip 0.9.6 [flutter]
- timeago 2.0.28
- url_launcher 5.7.10 [flutter url_launcher_platform_interface url_launcher_web url_launcher_linux url_launcher_macos url_launcher_windows]

dev dependencies:
- flutter_launcher_icons 0.8.1 [image args yaml path]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- _fe_analyzer_shared 7.0.0 [meta]
- analyzer 0.39.17 [_fe_analyzer_shared args charcode cli_util collection convert crypto glob html meta package_config path pub_semver source_span watcher yaml]
- archive 2.0.13 [crypto args path]
- args 1.6.0
- async 2.5.0-nullsafety.1 [collection]
- boolean_selector 2.1.0-nullsafety.1 [source_span string_scanner]
- characters 1.1.0-nullsafety.3
- charcode 1.2.0-nullsafety.1
- cli_util 0.2.0 [path]
- clock 1.1.0-nullsafety.1
- collection 1.15.0-nullsafety.3
- connectivity 0.4.9+2 [flutter meta connectivity_platform_interface connectivity_macos connectivity_for_web]
- connectivity_for_web 0.3.1+2 [connectivity_platform_interface flutter_web_plugins flutter]
- connectivity_macos 0.1.0+4 [flutter]
- connectivity_platform_interface 1.0.6 [flutter meta plugin_platform_interface]
- convert 2.1.1 [charcode typed_data]
- crypto 2.1.5 [collection convert typed_data]
- csslib 0.16.2 [source_span]
- dart_style 1.3.6 [analyzer args path source_span]
- fake_async 1.2.0-nullsafety.1 [clock collection]
- file 5.2.1 [intl meta path]
- firebase 7.3.0 [http http_parser js]
- firebase_analytics_platform_interface 1.0.3 [flutter meta]
- firebase_analytics_web 0.1.1 [flutter flutter_web_plugins firebase firebase_analytics_platform_interface meta]
- firebase_core_platform_interface 2.0.0 [flutter meta plugin_platform_interface quiver]
- firebase_core_web 0.2.1 [firebase_core_platform_interface flutter flutter_web_plugins meta js]
- firebase_crashlytics_platform_interface 1.1.3 [flutter meta collection firebase_core plugin_platform_interface]
- firebase_messaging_platform_interface 1.0.0-dev.5 [flutter meta firebase_core plugin_platform_interface]
- flutter_blurhash 0.5.0 [flutter meta]
- flutter_cache_manager 1.4.2 [flutter path_provider uuid http path sqflite pedantic clock file rxdart]
- flutter_local_notifications_platform_interface 2.0.0 [flutter plugin_platform_interface]
- flutter_plugin_android_lifecycle 1.0.9 [flutter]
- flutter_web_plugins 0.0.0 [flutter characters collection meta typed_data vector_math]
- glob 1.2.0 [async collection node_io path pedantic string_scanner]
- google_sign_in_platform_interface 1.1.2 [flutter meta quiver]
- google_sign_in_web 0.9.1+1 [google_sign_in_platform_interface flutter flutter_web_plugins meta js]
- gql 0.12.3 [source_span meta collection]
- graphql 3.1.0 [meta http mime path crypto http_parser uuid_enhanced gql rxdart websocket quiver]
- html 0.14.0+3 [csslib source_span]
- http_parser 3.1.4 [charcode collection source_span string_scanner typed_data]
- image 2.1.14 [archive xml]
- image_picker_platform_interface 1.1.0 [flutter meta http plugin_platform_interface]
- in_app_review_platform_interface 1.0.3 [flutter url_launcher package_info plugin_platform_interface]
- js 0.6.2
- list_ext 0.1.13 [quiver collection]
- matcher 0.12.10-nullsafety.1 [stack_trace]
- meta 1.3.0-nullsafety.3
- mime 0.9.7
- nested 0.0.4 [flutter]
- node_interop 1.1.1 [js]
- node_io 1.1.1 [node_interop path]
- octo_image 0.3.0 [flutter flutter_blurhash]
- package_config 1.9.3 [path charcode]
- path 1.8.0-nullsafety.1
- path_drawing 0.4.1+1 [vector_math meta path_parsing flutter]
- path_parsing 0.1.4 [vector_math meta]
- path_provider_linux 0.0.1+2 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 0.0.4+3 [flutter]
- path_provider_platform_interface 1.0.3 [flutter meta platform plugin_platform_interface]
- pedantic 1.9.0
- petitparser 3.0.4 [meta]
- platform 2.2.1
- plugin_platform_interface 1.0.3 [meta]
- process 3.0.13 [file intl meta path platform]
- pub_semver 1.4.4 [collection]
- quiver 2.1.3 [matcher meta]
- random_color 1.0.5 [flutter]
- rxdart 0.24.1
- shared_preferences_linux 0.0.2+2 [file flutter meta path path_provider_linux shared_preferences_platform_interface]
- shared_preferences_macos 0.0.1+10 [shared_preferences_platform_interface flutter]
- shared_preferences_platform_interface 1.0.4 [meta flutter]
- shared_preferences_web 0.1.2+7 [shared_preferences_platform_interface flutter flutter_web_plugins meta]
- sky_engine 0.0.99
- source_span 1.8.0-nullsafety.2 [charcode collection path term_glyph]
- sqflite 1.3.1+1 [flutter sqflite_common path]
- sqflite_common 1.0.2+1 [synchronized path meta]
- stack_trace 1.10.0-nullsafety.1 [path]
- stream_channel 2.1.0-nullsafety.1 [async]
- string_scanner 1.1.0-nullsafety.1 [charcode source_span]
- synchronized 2.2.0+2
- term_glyph 1.2.0-nullsafety.1
- test_api 0.2.19-nullsafety.2 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- timezone 0.5.9 [path]
- typed_data 1.3.0-nullsafety.3 [collection]
- url_launcher_linux 0.0.1+1 [flutter]
- url_launcher_macos 0.0.1+7 [flutter]
- url_launcher_platform_interface 1.0.9 [flutter meta plugin_platform_interface]
- url_launcher_web 0.1.5+1 [url_launcher_platform_interface flutter flutter_web_plugins meta]
- url_launcher_windows 0.0.1+3 [flutter]
- uuid 2.2.2 [crypto convert]
- uuid_enhanced 3.0.2 [crypto convert collection]
- vector_math 2.1.0-nullsafety.3
- watcher 0.9.7+15 [async path pedantic]
- websocket 0.0.5
- xdg_directories 0.1.0 [path process flutter]
- xml 4.2.0 [collection convert meta petitparser]
- yaml 2.2.1 [charcode collection string_scanner source_span]
```

</details>

---

**Comment by markusaksli-nc** [comment]
Hi @pawsong 
Could you try upgrading to the latest `firebase_messaging: ^8.0.0-dev.8` and seeing if that fixes it for you?
<details>
 <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 1.24.0-8.0.pre.177, on Microsoft Windows [Version 10.0.19041.572], locale et-EE)
    • Flutter version 1.24.0-8.0.pre.177 at C:\Development\flutter_master
    • Framework revision fb63371589 (11 hours ago), 2020-11-09 18:13:07 -0800
    • Engine revision d642a939c3
    • Dart version 2.12.0 (build 2.12.0-29.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.6858069\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.6858069
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Ultimate Edition (version 2020.2)
    • IntelliJ at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\202.7660.26
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (6 available)
    • SM G950F (mobile)           • ce12171c51cc001c03 • android-arm64  • Android 9 (API 28)
    • sdk gphone x86 arm (mobile) • emulator-5554      • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)           • windows            • windows-x64    • Microsoft Windows [Version 10.0.19041.572]
    • Web Server (web)            • web-server         • web-javascript • Flutter Tools
    • Chrome (web)                • chrome             • web-javascript • Google Chrome 86.0.4240.183
    • Edge (web)                  • edge               • web-javascript • Microsoft Edge 86.0.622.61

• No issues found!
```
</details>

I can't reproduce this.

**Comment by pawsong** [comment]
@markusaksli-nc Still happens.

**Comment by pawsong** [comment]
@markusaksli-nc Could you check if [this line (FlutterFirebaseMessagingUtils.java#L92)](https://github.com/FirebaseExtended/flutterfire/blob/7d91938fb9a78078f658ec099bc72be6ecd08773/packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingUtils.java#L92) is executed when you try to reproduce this issue?

The values of "titleLocArgs" and "bodyLocArgs" keys is a java native string array, which is not handled by flutter engine (as you see [here](https://github.com/flutter/engine/blob/48687ca3ca6e4f24281ea998789cc21ad840accb/shell/platform/android/io/flutter/plugin/common/StandardMessageCodec.java#L208))

**Comment by pawsong** [comment]
IMHO, we can use `List<String>` instead of pure java string array for values of "titleLocArgs" and "bodyLocArgs" to fix this issue. `List` seems to be supported by flutter engine. [link](https://github.com/flutter/engine/blob/48687ca3ca6e4f24281ea998789cc21ad840accb/shell/platform/android/io/flutter/plugin/common/StandardMessageCodec.java#L260)

**Comment by markusaksli-nc** [comment]
My bad, the notification payload wasn't following the FCM HTTP JSON format exatly. I can reproduce this on the latest flutter master `1.24.0-8.0.pre.194` with `firebase_messaging: ^8.0.0-dev.8` using the official example.
<details>
 <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 1.24.0-8.0.pre.194, on Microsoft Windows [Version 10.0.19041.572], locale et-EE)
    • Flutter version 1.24.0-8.0.pre.194 at C:\Development\flutter_master
    • Framework revision 018467cdb1 (2 hours ago), 2020-11-11 02:04:03 -0500
    • Engine revision 81f219c59c
    • Dart version 2.12.0 (build 2.12.0-31.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.6953283\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.6953283
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Ultimate Edition (version 2020.2)
    • IntelliJ at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\202.7660.26
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (5 available)
    • SM G950F (mobile) • ce12171c51cc001c03 • android-arm64  • Android 9 (API 28)
    • Windows (desktop) • windows            • windows-x64    • Microsoft Windows [Version 10.0.19041.572]
    • Web Server (web)  • web-server         • web-javascript • Flutter Tools
    • Chrome (web)      • chrome             • web-javascript • Google Chrome 86.0.4240.183
    • Edge (web)        • edge               • web-javascript • Microsoft Edge 86.0.622.63

• No issues found!
```
</details>
<details>
 <summary>Payload</summary>

```
{
  "notification": {
    "body": "this is a body",
    "title": "this is a title",
    "body_loc_args": [
      "a",
      "b"
    ]
  },
  "priority": "high",
  "to": "FCM Token"
}
```
</details>

**Comment by Salakar** [comment]
Fix pushed, will do a dev release after I've also fixed #4078
